app/services/code_climate/projects_summary_service.rb

```diff
@@ -38,12 +39,21 @@ def open_issues_count_average
     end
 
     def ratings
-      code_climate_ratings = code_climate_metrics.pluck(:code_climate_rate).compact
-      code_climate_ratings.each_with_object(Hash.new(0)) do |code_climate_rate, ratings|
-        ratings[code_climate_rate] += 1
+      @code_climate_ratings = code_climate_metrics.pluck(:id,
+                                                         :code_climate_rate,
+                                                         'code_climate_project_metrics.created_at')
+      @code_climate_ratings = sort_ratings_by_date
+      @code_climate_ratings.each_with_object(Hash.new(0)) do |code_climate_metric, ratings|
+        letter = code_climate_metric[1]
+        ratings[letter] += 1 if letter.present?
```

Comment:
```suggestion
      @ratings ||= begin
        sort_ratings_by_date(
          code_climate_metrics.pluck(:id, :code_climate_rate, 'code_climate_project_metrics.created_at')
        ).each_with_object(Hash.new(0)) do |code_climate_metric, ratings|
          letter = code_climate_metric[1]
          ratings[letter] += 1 if letter.present?
        end
```

I think something like that would look simpler to me

Reply:
notice that I pass the intermediate result as a param to `sort_ratings_by_date` and that you can call `ratings` directly here https://github.com/rootstrap/rs-code-review-metrics/pull/370/files#diff-229e40836c3086defd645ce882813e3ee45dcce4d1b88f760fc1000c72454cdfR102

Reply:
the thing is that in that line I need a hash with [id,rate] and `ratings` returns [rate,count]. I just refactored it to make it simpler, tell me what you think.